xapi/xapi.go

```diff
@@ -169,7 +209,11 @@ func Select(client kv.Client, req *tipb.SelectRequest, concurrency int, keepOrde
 	if resp == nil {
 		return nil, errors.New("client returns nil response")
 	}
-	result := &selectResult{resp: resp}
+	result := &selectResult{
+		resp:    resp,
+		results: make(chan PartialResult, 5),
+		done:    make(chan error, 1),
```

Comment:
Why not buffered channel ? Do we need to block fetch goroutine until wake up another goroutine ?